Mainnet/DividendToken/DividendToken/DividendToken.cs

```diff
@@ -0,0 +1,288 @@
+﻿using Stratis.SmartContracts;
+using Stratis.SmartContracts.Standards;
+
+[Deploy]
+public class DividendToken : SmartContract, IStandardToken
+{
+
+    public ulong Dividends
+    {
+        get => PersistentState.GetUInt64(nameof(this.Dividends));
+        private set => PersistentState.SetUInt64(nameof(this.Dividends), value);
+    }
+
+    private Account GetAccount(Address address) => PersistentState.GetStruct<Account>($"Account:{address}");
+
+    private void SetAccount(Address address, Account account) => PersistentState.SetStruct($"Account:{address}", account);
+
+
+    public DividendToken(ISmartContractState state, ulong totalSupply, string name, string symbol)
+        : base(state)
+    {
+        this.TotalSupply = totalSupply;
+        this.Name = name;
+        this.Symbol = symbol;
+        this.SetBalance(Message.Sender, totalSupply);
+    }
```

Comment:
`Decimals` also need to be added to IStandrdToken interface and set in the constructor of all new tokens. This is for display purposes so clients of the contract know how to properly display satoshi values.

Existing PR for the Decimal value in StandardToken contract here: https://github.com/stratisproject/CirrusSmartContracts/pull/17

Still needs a PR to IStandardToken interface in Stratis.SmartContracts.Standards

Reply:
I may update the code when that #17 approved but i will stick to current implementation of StandardToken for now.

Reply:
Looks like that PR is up to you and I to approve and merge, nobody else is looking at it and it's been there for months. The change matches ERC token standards and is needed for Opdex and general wallets for token display purposes. Not a good idea to have mainnet tokens without it.